package.json

```diff
@@ -16,6 +18,7 @@
     "@changesets/changelog-github": "^0.4.0",
     "@changesets/cli": "^2.0.0",
     "@types/react": "^18.0.0",
+    "markdown-tm-language": "github:wooorm/markdown-tm-language",
```

Comment:
are there plans to npm publish this?
Or no?

Reply:
I'm not really sure if it makes sense there. Nothing really consumes those files normally from packages, they all have to be wrapped up in extensions.

I'm planning to get things into other places, e.g., github/linguist, at which point they are pulled into starry-night.

Shiki also pulls things in from repos

Reply:
I don’t think semver there makes sense, there are never breaking changes, or, every change is a breaking change. As there is no API.
So basically every push to main there would result in a new release.
And what’s the benefit then?

Reply:
I do think there’s value in having something like this:
https://github.com/microsoft/vscode/blob/0d861e646719183875afccd4b649e7d604d06484/extensions/markdown-basics/syntaxes/markdown.tmLanguage.json#L2-L7.

Maybe licensing info too even. As such files are often copy/pasted around.

But I wanted to get this in here quickly so people can try it out!

Reply:
> I don’t think semver there makes sense, there are never breaking changes, or, every change is a breaking change. As there is no API.

It doesn't strictly need to be semver.
You could use the date at the tag.
Or skip tagging entirely and refer to a specific commit hash.

>  And what’s the benefit then?

knowing the exact version of the code used, without needing to copy and re-commit all the files here.

Reply:
> At that point, as this file is checked in, why not a "generate": "curl ... > packages/..." script, which someone can run once in a while?

The script could update the hash in the package.json, sure.

> Someone needs to remember, go to the repo, copy/paste the hash, and change that here.

Someone would need to know to run and remember to run the script as well 🤷 

> This is indeed useful, but should be possible to automate with current setup 

It could be possible.
When you say automate, do you mean just the script?
Or something like a GitHub action which automatically opens a PR to respond to changes?

Reply:
> The script could update the hash in the package.json, sure.

I mean instead of that. Not having a dep in package.json. No hash there.
Only crawling. Adding the hash in the crawled result.

> Someone would need to know to run and remember to run the script as well 🤷

Indeed, but not the other steps (going to a repo, copy current sha, pasting it here). I have a problem with those steps.

> When you say automate, do you mean just the script?

Something like `git ls-remote https://github.com/wooorm/markdown-tm-language` / `curl https://api.github.com/repos/wooorm/markdown-tm-language/branches/main`. Haven’t found an API that only returns the current hash.

Reply:
Added a script that adds the SHA, and can be run to crawl the current latest release, when someone is ready